.github/workflows/ci-core.yml

```diff
@@ -518,15 +526,41 @@ jobs:
         run: |
           make rm-mocked
           make generate
-      - name: Ensure clean after generate
-        run: |
-          git add --all
-          git diff --stat --cached --exit-code
-      - run: make gomodtidy
-      - name: Ensure clean after tidy
+
+      # If asked to fix issues, set up Git user and GitHub token to make the commit.
+      - name: Set up Git user to commit fixes
+        if: ${{ always() && contains( github.event.pull_request.labels.*.name, 'auto-fix') }}
+        shell: bash
         run: |
-          git add --all
-          git diff --minimal --cached --exit-code
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Setup GitHub Token to commit fixes
+        if: ${{ always() && contains( github.event.pull_request.labels.*.name, 'auto-fix') }}
+        id: setup-github-token
+        uses: smartcontractkit/.github/actions/setup-github-token@ef78fa97bf3c77de6563db1175422703e9e6674f # setup-github-token@0.1.2
+        with:
+          aws-role-arn: ${{ secrets.AWS_OIDC_CHAINLINK_CI_AUTO_PR_TOKEN_ISSUER_ROLE_ARN }}
+          aws-lambda-url: ${{ secrets.AWS_INFRA_RELENG_TOKEN_ISSUER_LAMBDA_URL }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-role-duration-seconds: "1800"
+
+      - name: Commit changes
```

Comment:
There's technically a race condition here between the `golangci-lint --fix` and this one.

Reply:
I mean that we’re going to checkout the same ref twice, and try and commit to the same branch concurrently. So the 1st commit will work, and the 2nd commit will fail because the local branch is now behind.

The concurrency group is at the workflow level I believe. But for a single workflow run, there’s two jobs committing.

Reply:
Actually because linting is a matrix, we can technically have >8 jobs trying to simultaneously commit to the same branch.